docs/source/storage.md

```diff
@@ -0,0 +1,262 @@
+# torch.Storage
+
+In PyTorch, a regular tensor is a multi-dimensional array that is defined by the following components:
+
+- Storage: The actual data of the tensor, stored as a contiguous, one-dimensional array of bytes.
+- `dtype`: The data type of the elements in the tensor, such as torch.float32 or torch.int64.
+- `shape`: A tuple indicating the size of the tensor in each dimension.
+- Stride: The step size needed to move from one element to the next in each dimension.
+- Offset: The starting point in the storage from which the tensor data begins. This will usually be 0 for newly
+  created tensors.
+
+These components together define the structure and data of a tensor, with the storage holding the
+actual data and the rest serving as metadata.
+
+## Untyped Storage API
+
+A {class}`torch.UntypedStorage` is a contiguous, one-dimensional array of elements. Its length is equal to the number of
+bytes of the tensor. The storage serves as the underlying data container for tensors.
+In general, a tensor created in PyTorch using regular constructors such as {func}`~torch.zeros`, {func}`~torch.zeros_like`
+or {func}`~torch.Tensor.new_zeros` will produce tensors where there is a one-to-one correspondence between the tensor
+storage and the tensor itself.
+
+However, a storage is allowed to be shared by multiple tensors.
+For instance, any view of a tensor (obtained through {meth}`~torch.Tensor.view` or some, but not all, kinds of indexing
+like integers and slices) will point to the same underlying storage as the original tensor.
+When serializing and deserializing tensors that share a common storage, the relationship is preserved, and the tensors
+continue to point to the same storage. Interestingly, deserializing multiple tensors that point to a single storage
+can be faster than deserializing multiple independent tensors.
+
+A tensor storage can be accessed through the {meth}`~torch.Tensor.untyped_storage` method. This will return an object of
+type {class}`torch.UntypedStorage`.
+Fortunately, storages have a unique identifier accessed through the {meth}`torch.UntypedStorage.data_ptr` method.
+In regular settings, two tensors with the same data storage will have the same storage `data_ptr`.
+However, tensors themselves can point to two separate storages, one for its data attribute and another for its grad
+attribute. Each will require a `data_ptr()` of its own. In general, there is no guarantee that a
+{meth}`torch.Tensor.data_ptr` and {meth}`torch.UntypedStorage.data_ptr` match and this should not be assumed to be true.
+
+Untyped storages are somewhat independent of the tensors that are built on them. Practically, this means that tensors
+with different dtypes or shape can point to the same storage.
+It also implies that a tensor storage can be changed, as the following example shows:
+
+> ```python
+> >>> t = torch.ones(3)
```

Comment:
Something is wrong with the formatting here. Can you please remove the leading > sign?